incubator/hnc/docs/user-guide/how-to.md

```diff
@@ -421,3 +499,86 @@ spec:
+| `hnc/reconcilers/hierconfig/concurrent_peak`         | The peak concurrent HC reconciliations happened in the past 60s, which is also the minimum Stackdriver reporting period and the one we're using |
+| `hnc/reconcilers/hierconfig/hierconfig_writes_total` | The number of HC writes happened during HC reconciliations |
+| `hnc/reconcilers/hierconfig/namespace_writes_total`  | The number of namespace writes happened during HC reconciliations |
+| `hnc/reconcilers/object/total`                       | The total number of object reconciliations happened |
+| `hnc/reconcilers/object/concurrent_peak`             | The peak concurrent object reconciliations happened in the past 60s, which is also the minimum Stackdriver reporting period and the one we're using |
+
+#### Use Stackdriver on GKE
+
+To view HNC Metrics in Stackdriver, you will need a GKE cluster with HNC
+installed and a method to access Cloud APIs, specifically Stackdriver monitoring
+APIs, from GKE. We recommend using [Workload
+Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity)
+to minimize the permissions required to log metrics.  Once it's set up, you can
+view the metrics in Stackdriver  [Metrics
+Explorer](https://cloud.google.com/monitoring/charts/metrics-explorer) by
+searching the metrics keywords (e.g. `namespace_conditions`).
+
+In order to monitor metrics via Stackdriver:
+1. Enable Workload Identity (WI) on either a
+   [new](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_workload_identity_on_a_new_cluster)
+   or
+   [existing](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_workload_identity_on_an_existing_cluster)
+   cluster.
+2. Install HNC as described [above](#admin-install).
+4. [Create a Google service account (GSA)](https://cloud.google.com/docs/authentication/production#creating_a_service_account):
+    ```bash
+    gcloud iam service-accounts create [GSA_NAME]
+    ```
+5. Grant “[Monitoring Metric Writer](https://cloud.google.com/monitoring/access-control#mon_roles_desc)”
+role to the GSA:
+    ```bash
+    gcloud projects add-iam-policy-binding [PROJECT_ID] --member \
+      "serviceAccount:[GSA_NAME]@[PROJECT_ID].iam.gserviceaccount.com" \
+      --role "roles/monitoring.metricWriter"
+    ```
+6. Create an [Cloud IAM policy binding](https://cloud.google.com/sdk/gcloud/reference/iam/service-accounts/add-iam-policy-binding)
+between `hnc-system/default` KSA and the newly created GSA:
+     ```
+     gcloud iam service-accounts add-iam-policy-binding \
+       --role roles/iam.workloadIdentityUser \
+       --member "serviceAccount:[PROJECT_ID].svc.id.goog[hnc-system/default]" \
+       [GSA_NAME]@[PROJECT_ID].iam.gserviceaccount.com
+   ```
+6. Add the `iam.gke.io/gcp-service-account=[GSA_NAME]@[PROJECT_ID]` annotation to
+the KSA, using the email address of the Google service account:
```

Comment:
Seems the numbers are not right here, 2.3.4.5.6?

Reply:
Fixed!